popup/popup.js

```diff
@@ -1,248 +1,366 @@
+      document.querySelector('#twitch_button').textContent =
+        'Deconnect from twitch'
+      document
+        .querySelector('#twitch_button')
+        .setAttribute('action', 'twitch_deconnect')
+    } else {
+      document.querySelector('#twitch_channel').disabled = false
+      document.querySelector('#twitch_button').textContent = 'Connect to twitch'
+      document
+        .querySelector('#twitch_button')
+        .setAttribute('action', 'twitch_connect')
+    }
+  }
+)
 
-function update_action_pending(message) {
-	if (!message?.obj?.length) return
-	const pending = document.querySelector("#popup-ide-actions-pending")
-	pending.innerHTML = ''
-	const span = document.createElement('span')
-	span.textContent = 'Pending actions'
-	pending.appendChild(span)
-	const ul = document.createElement('ul')
-	pending.appendChild(ul)
-	for (action of message.obj) {
-		const li = document.createElement('li')
-		li.textContent = JSON.stringify(action)
-		ul.appendChild(li)
-	}
+function update_cg_id (message) {
+  console.log("show CG id");
+  document.querySelector('#identifiers').innerHTML =
+    message.obj
+      ? `${message.obj.pseudo} (${message.obj.userId}) - ${message.obj.publicHandle}`
+      : 'Codingame identifiers not found'
 }
-browser.runtime.sendMessage({
-	command: 'action_get_buffer',
-}, r => {
-	log('DEBUG', r)
-	update_action_pending(r)
+browser.runtime.sendMessage({ command: 'get_cg_id' }, r => {
+  log('DEBUG', r)
+  update_cg_id(r)
 })
 
-browser.runtime.onMessage.addListener(function messageListener(message, sender, reply) {
-	log('debug', ['onMessage', message, sender, reply])
-	if (message.command === 'buffer_change') {
-		update_action_pending(message)
-	} else {
-		log('ERROR', 'unkown command message')
-	}
-})
+function update_tournaments () {
+  console.log('update_tournaments')
+  browser.runtime.sendMessage({ command: 'get_tournament_infos' }, r => {
+    console.log('get_tournament_infos response :', r)
+    if (r.res === 'obj') {
+      const active = !!r.obj?.current_tournament?.id
+      const joined = !!r.obj?.joined_tournament?.id
+      document.querySelector('#search_tournament_button').disabled = active
+      document.querySelector('#join_tournament_button').disabled =
+        joined || !active
+    }
+  })
+}
+update_tournaments()
 
-document.querySelector("#popup-load").classList.add("hidden");
+function update_action_pending (message) {
+  if (!message?.obj?.length) return
+  const pending = document.querySelector('#popup-ide-actions-pending')
+  pending.innerHTML = ''
+  const span = document.createElement('span')
+  span.textContent = 'Pending actions'
+  pending.appendChild(span)
+  const ul = document.createElement('ul')
+  pending.appendChild(ul)
+  for (action of message.obj) {
+    const li = document.createElement('li')
+    const actionName = Object.keys(action)[0];
+    const actionValue = Object.values(action)[0];
+    li.textContent = actionName + ": " + actionValue;
```

Comment:
Il peut y avoir plusieurs clé/valeur (ex avec block_letter)
```suggestion
    li.textContent = Object.entries(action).map(([k, v]) => `${k}: ${v}`).join(', ');
```